yarn-project/key-store/src/secp256k1/test_key_store.ts

```diff
@@ -0,0 +1,128 @@
+  // But the inputs to our circuits use AztecAddress, so for now, we will just use AztecAddress
+  // as a placeholder for an Ethereum address. Longer term, we want to replace AztecAddress with EthAddress
+  // in classes like TxRequest/CallContext/etc.
+
+  constructor(private secp256k1: Secp256k1, private ecdsa: Ecdsa) {}
+
+  /**
+   * Adds a new account to the TestSecp256k1KeyStore with a randomly generated ConstantSecp256k1KeyPair.
+   * The account will have its own private and public key pair, which can be used for signing transactions.
+   *
+   * @returns A promise that resolves to the newly created account's AztecAddress.
+   */
+  public addAccount() {
+    const keyPair = ConstantSecp256k1KeyPair.random(this.secp256k1, this.ecdsa);
+    this.accounts.push(keyPair);
+    return Promise.resolve(keyPair.getPublicKey().toAztecAddress());
+  }
+
+  /**
+   * Retrieves the public addresses of all accounts stored in the TestSecp256k1KeyStore.
+   * The returned addresses are instances of `AztecAddress` and can be used for subsequent operations
+   * such as signing transactions or fetching public/private keys.
+   *
+   * @returns A Promise that resolves to an array of AztecAddress instances.
+   */
+  public getAccounts() {
+    return Promise.resolve(this.accounts.map(a => a.getPublicKey().toAztecAddress()));
+  }
+
+  /**
+   * Retrieves the private key of the account associated with the specified AztecAddress.
+   * Throws an error if the provided address is not found in the list of registered accounts.
+   *
+   * @param address - The AztecAddress instance representing the account for which the private key is requested.
+   * @returns A Promise that resolves to a Buffer containing the private key.
+   */
+  public getAccountPrivateKey(address: AztecAddress): Promise<Buffer> {
+    const account = this.getAccount(address);
+    return account.getPrivateKey();
+  }
+
+  /**
+   * Retrieve the public key of an account with a given address.
+   * Searches for the corresponding account in the accounts array, and returns its public key.
+   * If the account is not found, an error is thrown.
+   *
+   * @param address - The AztecAddress of the account whose public key is to be retrieved.
+   * @returns A Promise that resolves with the Point instance representing the public key of the account.
+   */
+  public getAccountPublicKey(address: AztecAddress): Promise<EthPublicKey> {
+    const account = this.getAccount(address);
+    return Promise.resolve(account.getPublicKey());
+  }
+
+  /**
+   * Retrieves an array of public keys for all accounts stored in the TestSecp256k1KeyStore.
+   * These public keys can be used for verifying signatures on transactions and messages.
+   *
+   * @returns A promise that resolves to an array of public keys associated with the accounts in the TestSecp256k1KeyStore.
+   */
+  public getSigningPublicKeys() {
+    return Promise.resolve(this.accounts.map(a => a.getPublicKey()));
+  }
```

Comment:
Why `getSigningPublicKeys` vs `AccountPublicKey`?

Reply:
It's because `getSigningPublicKeys` was meant originally to get the signing key but we didn't really have signing keys until the ECDSA work. I don't want to change the key store interface so I am not changing this now, we can come back to rename this later (because the key store interface itself might change with the discussions around key management).